src/api/syscall.c

```diff
@@ -588,8 +588,13 @@ static inline void mcsPreemptionPoint(irq_t irq)
     } else if (NODE_STATE(ksCurSC)->scRefillMax) {
         /* otherwise, if the thread is not schedulable, the SC could be valid - charge it if so */
         chargeBudget(NODE_STATE(ksConsumed), false, CURRENT_CPU_INDEX(), true);
+    } else {
```

Comment:
When I last considered this code path with Mitch, we concluded that the SC being deleted would always result in commitTime being called due to the reschedule of a new thread.  ksConsumed will be set to 0 then.  I agree that this doesn't lead to a violation of budget because the SC must have had enough budget to get to the next preemption point.  (Although if there was an overrun then the overrun amount wouldn't be accounted to anything...).

So is this change just to make verification easier?

Reply:
This is to avoid having the new SC charged for budget from the previous task, which could include the entire execution of that task before entering the kernel. Charging that time to another task would break temporal isolation.

Reply:
But it wouldn't because commitTime sets ksConsumed to 0 unconditionally. (https://github.com/seL4/seL4/blob/master/include/kernel/thread.h#L158)